src/azure-cli/azure/cli/command_modules/cdn/tests/latest/test_waf_scenarios.py

```diff
@@ -0,0 +1,447 @@
+                                self.kwargs['block_response_status_code']),
+                  JMESPathCheck('policySettings.defaultCustomBlockResponseBody', self.kwargs['block_response_body']),
+                  JMESPathCheck('policySettings.mode', PolicyMode.prevention.value),
+                  JMESPathCheck('tags', {}),
+                  JMESPathCheck('endpointLinks', [])]
+        self.cmd('cdn waf policy set -g {resource_group} '
+                 '--name {policy} '
+                 '--sku Standard_Microsoft '
+                 '--mode Prevention '
+                 '--block-response-body {block_response_body} '
+                 '--block-response-status-code {block_response_status_code} '
+                 '--redirect-url {redirect_url} '
+                 '--tags="" '
+                 '--disabled',
+                 checks=checks)
+        self.cmd('cdn waf policy show -g {resource_group} -n {policy}', checks=checks)
+
+        # Remove managed rule group override
+        self.cmd('cdn waf policy managed-rule-set rule-group-override delete -g {resource_group} --policy-name {policy} '
+                 '--rule-set-type {rule_set_type} --rule-set-version {rule_set_version} -n {rule_group}')
+        with self.assertRaises(CLIError):
+            self.cmd('cdn waf policy managed-rule-set rule-group-override show -g {resource_group} '
+                     '--policy-name {policy} --rule-set-type {rule_set_type} '
+                     '--rule-set-version {rule_set_version} -n {rule_group}')
+
+        # Remove managed rule set
+        self.cmd('cdn waf policy managed-rule-set remove -g {resource_group} --policy-name {policy} '
+                 '--rule-set-type {rule_set_type} --rule-set-version {rule_set_version}')
+        with self.assertRaises(CLIError):
+            self.cmd('cdn waf policy managed-rule-set show -g {resource_group} --policy-name {policy} '
+                     '--rule-set-type {rule_set_type} --rule-set-version {rule_set_version}')
+
+        # Remove custom rule
+        self.cmd('cdn waf policy custom-rule delete -g {resource_group} --policy-name {policy} -n {custom_rule_1}')
+        with self.assertRaises(CLIError):
+            self.cmd('cdn waf policy custom-rule show -g {resource_group} --policy-name {policy} -n {custom_rule_1}')
+
+        # Remove rate limit rule
+        self.cmd('cdn waf policy rate-limit-rule delete -g {resource_group} --policy-name {policy} -n {rate_limit_rule_1}')
+        with self.assertRaises(CLIError):
+            self.cmd('cdn waf policy rate-limit-rule show -g {resource_group} --policy-name {policy} -n {rate_limit_rule_1}')
+
+        # Delete policy
+        self.cmd('cdn waf policy delete -g {resource_group} -n {policy}')
+
+
+class CdnWafEndpointLinkScenarioTest(CdnScenarioMixin, ScenarioTest):
+    @ResourceGroupPreparer(name_prefix='cli_test_cdn_domain')
+    def test_cdn_waf_endpoint_link(self, resource_group):
```

Comment:
This is not related with this PR. right?

Reply:
This is related - this tests adding a CDN WAF policy to a CDN endpoint, which is new functionality I've added in this PR.